core/libp2p/peer/peer_identity.hpp

```diff
@@ -0,0 +1,89 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef KAGOME_PEER_IDENTITY_HPP
+#define KAGOME_PEER_IDENTITY_HPP
+
+#include <memory>
+#include <string>
+#include <string_view>
+
+#include <outcome/outcome.hpp>
+#include "libp2p/multi/multiaddress.hpp"
+#include "libp2p/peer/peer_info.hpp"
+
+namespace libp2p::peer {
+  /**
+   * Identity of the given peer; includes its ID and multiaddress; usually is
+   * passed over the network
+   */
+  class PeerIdentity {
```

Comment:
PeerIdentity actually valid Multiaddr, so maybe we should extend from Multiaddress?